e2etest/bgp_test.go

```diff
@@ -94,69 +97,66 @@ var _ = ginkgo.Describe("BGP", func() {
 		var allNodes *corev1.NodeList
-		ginkgo.BeforeEach(func() {
-			configData := configFile{
-				Pools: []addressPool{
-					{
-						Name:     "bgp-test",
-						Protocol: BGP,
-						Addresses: []string{
-							"192.168.10.0/24",
-							"fc00:f853:0ccd:e799::/124",
-						},
+		configData := configFile{
+			Pools: []addressPool{
+				{
+					Name:     "bgp-test",
+					Protocol: BGP,
+					Addresses: []string{
+						"192.168.10.0/24",
+						"fc00:f853:0ccd:e799::/124",
 					},
 				},
-				Peers: []peer{
-					{
-						Addr:  frrContainerIPv4,
-						ASN:   64512,
-						MyASN: 64512,
-					},
+			},
+			Peers: []peer{
+				{
+					Addr:  *frrContainerIP,
+					ASN:   64512,
+					MyASN: 64512,
 				},
-			}
-			err := updateConfigMap(cs, configData)
-			framework.ExpectNoError(err)
+			},
+		}
+		err := updateConfigMap(cs, configData)
+		framework.ExpectNoError(err)
 
-			pairExternalFRRWithNodes(cs)
+		pairExternalFRRWithNodes(cs, version)
 
-			allNodes, err = cs.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
-			framework.ExpectNoError(err)
-		})
+		allNodes, err = cs.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+		framework.ExpectNoError(err)
 
-		ginkgo.It("should work for ExternalTrafficPolicy=Cluster", func() {
-			svc, _ := createServiceWithBackend(cs, f.Namespace.Name, corev1.ServiceExternalTrafficPolicyTypeCluster)
+		ginkgo.By("should work for ExternalTrafficPolicy=Cluster")
+		svc, _ := createServiceWithBackend(cs, f.Namespace.Name, corev1.ServiceExternalTrafficPolicyTypeCluster, version)
 
-			defer func() {
-				err := cs.CoreV1().Services(svc.Namespace).Delete(context.TODO(), svc.Name, metav1.DeleteOptions{})
-				framework.ExpectNoError(err)
-			}()
+		defer func() {
+			err := cs.CoreV1().Services(svc.Namespace).Delete(context.TODO(), svc.Name, metav1.DeleteOptions{})
+			framework.ExpectNoError(err)
+		}()
 
-			validateService(cs, svc, allNodes.Items)
-		})
+		validateService(cs, svc, allNodes.Items)
 
-		ginkgo.It("should work for ExternalTrafficPolicy=Local", func() {
-			svc, jig := createServiceWithBackend(cs, f.Namespace.Name, corev1.ServiceExternalTrafficPolicyTypeLocal)
-			err := jig.Scale(2)
-			framework.ExpectNoError(err)
+		ginkgo.By("should work for ExternalTrafficPolicy=Local")
+		svc, jig := createServiceWithBackend(cs, f.Namespace.Name, corev1.ServiceExternalTrafficPolicyTypeLocal, version)
+		err = jig.Scale(2)
+		framework.ExpectNoError(err)
 
-			epNodes, err := jig.ListNodesWithEndpoint() // Only nodes with an endpoint should be advertising the IP
-			framework.ExpectNoError(err)
+		epNodes, err := jig.ListNodesWithEndpoint() // Only nodes with an endpoint should be advertising the IP
+		framework.ExpectNoError(err)
 
-			defer func() {
-				err := cs.CoreV1().Services(svc.Namespace).Delete(context.TODO(), svc.Name, metav1.DeleteOptions{})
-				framework.ExpectNoError(err)
-			}()
+		defer func() {
```

Comment:
Here (and above) you are changing the scope of the defer to the outside context.